radio/src/gui/colorlcd/widgets/modelbmp.cpp

```diff
@@ -99,13 +113,19 @@ class ModelBitmapWidget: public Widget
         }
 
         if (rect.h >= 96 && rect.w >= 120) {
-          buffer->drawScaledBitmap(bitmap.get(), 0, 38, width(), height() - 38);
+          buffer->drawScaledBitmap(bitmap.get(), 0, 0, width(), height() - 38);
         } else {
           buffer->drawScaledBitmap(bitmap.get(), 0, 0, width(), height());
         }
       }
     }
 };
 
-BaseWidgetFactory<ModelBitmapWidget> modelBitmapWidget("ModelBmp", nullptr);
-const WidgetFactory * defaultWidget = &modelBitmapWidget;
+const ZoneOption ModelBitmapWidget::options[] = {
+    {STR_COLOR, ZoneOption::Color, OPTION_VALUE_UNSIGNED(DEFAULT_COLOR)},
+    {STR_SIZE, ZoneOption::TextSize, OPTION_VALUE_UNSIGNED(3)},
+    {STR_FILL_BACKGROUND, ZoneOption::Bool, OPTION_VALUE_BOOL(false)},
+    {STR_BG_COLOR, ZoneOption::Color, OPTION_VALUE_UNSIGNED(DEFAULT_BGCOLOR)},
+    {nullptr, ZoneOption::Bool}};
+
+BaseWidgetFactory<ModelBitmapWidget> modelBitmapWidget("ModelBmp", ModelBitmapWidget::options);
```

Comment:
> 
> 
> > option defaults doesn't seem to be importing colours... ie. DEFAULT_COLOR - it was a light blue, but option defaults to black
> 
> In the following instantiation, a `nullptr` is given for the option list. If you would provide one instead, it should give you what you're looking for.
> 
> https://github.com/EdgeTX/edgetx/blob/3eeca550dad17b174848f9bea59a0e5b8d45b9d5/radio/src/gui/colorlcd/widgets/modelbmp.cpp#L110

Isn't that what I did here though? The widget originally had a nullptr as it had no options, so I added it based on how the others were formatted.

Reply:
Ok, then we have a bug there ;-) Let's try to see what is wrong in the implementation of the widget factory. The code is supposed to init the values to their default options here:
https://github.com/EdgeTX/edgetx/blob/3cb496a477e3e3c36dfb96e0ab18035d1e7e38e7/radio/src/gui/colorlcd/widget.h#L142

Reply:
No, the bug is mine... :facepalm:

Reply:
It helps if you give valid values as the defaults 😆